tools/SwiftSyntax/SourceLength.swift

```diff
@@ -0,0 +1,87 @@
+//===------------------ SourceLength.swift - Source Length ----------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// The length a syntax node spans in the source code. From any AbsolutePosition
+/// you reach a node's end location by either adding its UTF-8 length or by
+/// inserting `lines` newlines and then moving `columns` columns to the right.
+public final class SourceLength {
+  public let lines: Int
+  public let columns: Int
```

Comment:
I feel naming `newlines` and (`columnsAtLastLine` or `remainingColumns`) is better fit here.
Or, `newlineCount` and (`columnCountAtLastLine` or `remainingColumnCount`)

Reply:
That's definitely an improvement. Thanks.